hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/BackgroundService.java

```diff
@@ -103,8 +105,21 @@ public void runPeriodicalTaskNow() throws Exception {
   }
 
   // start service
-  public void start() {
-    exec.scheduleWithFixedDelay(service, 0, interval, unit);
+  public synchronized void start() {
+    scheduledHandle = exec.scheduleWithFixedDelay(service, 0, interval, unit);
```

Comment:
Should we also add a check to not start the thread if scheduledHandle is not null?
cc @sumitagrawl